daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Classify.scala

```diff
@@ -5,37 +5,49 @@ package com.daml.lf
 package speedy
 
 import com.daml.lf.speedy.Speedy.{Control, Machine}
+import com.daml.scalautil.Statement.discard
 
 import scala.collection.mutable
 
 private[speedy] object Classify { // classify the machine state w.r.t what step occurs next
 
-  final class Counts(
-      var ctrlExpr: Int = 0,
-      var ctrlValue: Int = 0,
```

Comment:
Oddly, this `var` is never updated!?

Reply:
Wasn't `ctrlValue` was being updated on line 32 that you deleted?

Reply:
Oops! Quite correct. Let me generate a Pr to patch that.

Reply:
https://github.com/digital-asset/daml/pull/15071 fixes this issue